api/test/wfl/integration/modules/sg_test.clj

```diff
@@ -0,0 +1,145 @@
+(ns wfl.integration.modules.sg-test
+  (:require [clojure.test :refer [deftest testing is] :as clj-test]
+            [clojure.string :as str]
+            [wfl.environments :as env]
+            [wfl.module.batch :as batch]
+            [wfl.service.cromwell :refer [wait-for-workflow-complete submit-workflows]]
+            [wfl.tools.endpoints :as endpoints]
+            [wfl.tools.fixtures :as fixtures]
+            [wfl.tools.workloads :as workloads]
+            [wfl.util :as util :refer [absent? make-options]]
+            [wfl.module.sg :as sg])
+  (:import (java.time OffsetDateTime)
+           (java.util UUID)))
+
+(clj-test/use-fixtures :once fixtures/temporary-postgresql-database)
+
+(defn ^:private make-sg-workload-request []
+  (-> (UUID/randomUUID)
+      workloads/sg-workload-request
+      (assoc :creator (:email @endpoints/userinfo))))
+
+(defn mock-submit-workload [{:keys [workflows]} _ _ _ _ _]
+  (let [now       (OffsetDateTime/now)
+        do-submit #(assoc % :uuid (UUID/randomUUID)
+                          :status "Submitted"
+                          :updated now)]
+    (map do-submit workflows)))
+
+(deftest test-create-workload!
+  (letfn [(verify-workflow [workflow]
+            (is (absent? workflow :uuid))
+            (is (absent? workflow :status))
+            (is (absent? workflow :updated)))
+          (go! [workload-request]
+            (let [workload (workloads/create-workload! workload-request)]
+              (is (:created workload))
+              (is (absent? workload :started))
+              (is (absent? workload :finished))
+              (run! verify-workflow (:workflows workload))))]
+    (testing "single-sample workload-request"
+      (go! (make-sg-workload-request)))))
+
+(deftest test-create-workload-with-common-inputs
+  (let [common-inputs {:biobambam_bamtofastq.max_retries 2
+                       :ref_pac "gs://fake-location/temp_references/gdc/GRCh38.d1.vd1.fa.pac"}]
+    (letfn [(go! [inputs]
+              (letfn [(value-equal? [key] (= (key common-inputs) (key inputs)))]
+                (is (value-equal? :biobambam_bamtofastq.max_retries))
+                (is (value-equal? :ref_pac))))]
+      (run! (comp go! :inputs) (-> (make-sg-workload-request)
+                                   (assoc-in [:common :inputs] common-inputs)
+                                   workloads/create-workload!
+                                   :workflows)))))
+
+(deftest test-start-workload!
+  (letfn [(go! [workflow]
+            (is (:uuid workflow))
+            (is (:status workflow))
+            (is (:updated workflow)))]
+    (with-redefs-fn {#'batch/submit-workload! mock-submit-workload}
+      #(-> (make-sg-workload-request)
+           workloads/create-workload!
+           workloads/start-workload!
+           (as-> workload
+                 (is (:started workload))
+             (run! go! (:workflows workload)))))))
```

Comment:
These tests are all grabbed from `xx_test.clj`. This is testing that, if we mock submission to cromwell, we can create and start workloads. It is testing that we correctly insert the "cromwell" uuid, status, and the updated time

Reply:
The lambda here is being called by `with-redefs-fn`, it is just a lambda so that `with-redefs-fn` can rebind the vars inside that specific context (I think)

Reply:
It doesn't seem like I can leave docstrings on tests--do you think it would be helpful to annotate these with comments?